### ISSUE TITLE
Question about frame coordinate and openMVG_main_ComputeStructureFromKnownPoses

### ISSUE THREAD

**Issue by liming0791** [openMVG/openMVG#1420]
With known initial poses from VINS-MONO (VIO System), processing steps as:

1. openMVG_main_SfMInit_ImageListing
2. use python scripts to fill sfm_data.json (produced by step1) with known extrinsics (from VIO), and change the Intrinsic and distortion parameters. The rotation and center is carefully set as it should be.
3. openMVG_main_ComputeFeatures
4. openMVG_main_ComputeStructureFromKnownPoses

The poses in result.ply looks good, but the reconstructed points looks bad. And following OpenMVS steps failed.

However, openMVG_main_IncrementalSfM works fine and the result is ok with OpenMVS steps.

I wonder whether the problem is the coordinate difference between VINS-MONO and OpenMVG. So 

what is the frame coordinate set up in OpenMVG ? And any other thing could be wrong ?

**Comment by pmoulon** [comment]
Hi,

What I would do to investigate this:
1. Check the camera orientation with `openMVG_main_ExportCameraFrustums` -> https://github.com/openMVG/openMVG/blob/master/src/software/SfM/export/main_ExportCameraFrustums.cpp. I would see from here if the camera are pointing in the right direction.

2. I would check that some valid features are founds and some viable matches are found. It seems so in your case, since `openMVG_main_IncrementalSfM` work fine.

3. Can you tell me which parameter are you using on `openMVG_main_ComputeStructureFromKnownPoses`. Please note that there is two triangulation mode. The default one that will build track triplets and try to triangulate them. Or the triangulation of the matche file, that should react better in your case if the camera pose is approximate.

4. Please note that now OpenMVG provide `main_IncrementalSfM2`. This version can use a sfm_data file that already contains camera poses. It will triangulate the feature matches and continue to test to add the missing image if there is any.

FYI:
-Cameras should point with Z front and Y up.

Feel free to share some data here to debug this with the community ;-)

**Comment by liming0791** [comment]
@pmoulon 

Hi, 

Thanks for replying.

1. I used `openMVG_main_ExportCameraFrustums` to export cameras. However, the scale of the frustums is too big and look almost a mess. 
![screenshot from 2018-11-09 15-49-35](https://user-images.githubusercontent.com/4964570/48249991-bec95580-e437-11e8-8e0f-de4abf23f60b.png)

2. About parameters using on `openMVG_main_ComputeStructureFromKnownPoses`. The default one produce reasonable poses (position seem like) but poor structure. Triangulation of the match file produce almost nothing, and the parameters are `"-i", "reconstruction_dir/sfm_data.json", "-m", "matches_dir", "-f", "matches_dir/matches.f.bin", "-o", "reconstruction_dir/robust.bin", "-b", "-d", "-r", "1"`.

3. I used `main_IncrementalSfM2`, and also produced almost nothing.

4. Camera point with Z front and Y up, and X left or right ? I am sure that in VINS-MONO (VIO), camera point with Z front and Y down and X right.
